riscv/vector_unit.cc

```diff
@@ -86,6 +86,54 @@ template<class T> T& vectorUnit_t::elt(reg_t vReg, reg_t n, bool UNUSED is_write
   return regStart[n];
 }
 
+// The logic differences between 'elt()' and 'elt_group()' come from
+// the fact that, while 'elt()' requires that the element is fully
+// contained in a single vector register, the element group may span
+// multiple registers in a single register group (LMUL>1).
+//
+// Notes:
+// - We do NOT check that a single element - i.e., the T in the element
+//   group type std::array<T, N> - fits within a single register, or that
+//   T is smaller or equal to VSEW. Implementations of the instructions
+//   sometimes use a different T than what the specification suggests.
+//   Instructon implementations should 'require()' what the specification
+//   dictates.
+// - We do NOT check that 'vReg' is a valid register group, or that
+//   'n+1' element groups fit in the register group 'vReg'. It is
+//   the responsibility of the caller to validate those preconditions.
+template<typename EG> EG&
+vectorUnit_t::elt_group(reg_t vReg, reg_t n, bool UNUSED is_write) {
+#ifdef WORDS_BIGENDIAN
+# error This logic is invalid for Big Endian mode.
+#endif
```

Comment:
This is problematic since we have big-endian users.  Two options:
- The obvious one, make this logic correct for big-endian hosts.
- Keep the old big-endian functionality working, and disallow use of this functionality on big-endian hosts.  You can have something like

```
#ifdef WORDS_BIGENDIAN
  abort();
#endif
```

and then also change the `isa_parser` to disallow any extensions that rely on this when `WORDS_BIGENDIAN` is defined.

Reply:
Done, thanks for the suggestion.